### PR TITLE
added wrapper functions for store operations

### LINKED ISSUE #856: glusterd2 is not responding if ETCD server is down
even if  ETCD service is down, glusterd2 should return an internal server error(500)

**steps to reproduce issue** 

1)create a volume
2)stop external ETCD server
3get volume list

**Expected result:**
Internal Server Error (500)

**Actual result:**
1)For curl request, there is no response from glusterd2
2)Response from rest client

```
Error: Post http://10.70.43.235:24007/v1/volumes: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

### PR DESCRIPTION
**Pull request by Madhu-1**
Fixes #856 

Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@prashanthpai @aravindavk @kshlm  please review

File: glusterd2/peer/store-utils.go [hunk 1]
```diff
@@ -257,7 +257,10 @@ func GetPeerByAddrs(addrs []string) (*Peer, error) {
 func GetPeerIDByAddr(addr string) (uuid.UUID, error) {
 	p, e := GetPeerByAddrF(addr)
 	if e != nil {
-		return nil, errors.ErrPeerNotFound
+		if e == errors.ErrPeerNotFound {
```
Comment (prashanthpai) [inline glusterd2/peer/store-utils.go]: You can just return error as is here.

```go
if e != nil {
    return nil, e
}
```

File: glusterd2/store/store.go [hunk 2]
```diff
@@ -18,7 +19,10 @@ import (
 )
 
 const (
-	sessionTTL = 30 // used for etcd mutexes and liveness key
+	sessionTTL    = 30              // used for etcd mutexes and liveness key
+	getTimeout    = 5 * time.Second //timeout for store Get operation
```
Comment (prashanthpai) [inline glusterd2/store/store.go]: The variable names are self-explanatory, you can remove the comments for timeouts.

File: glusterd2/store/store.go [hunk 3]
```diff
@@ -177,3 +181,37 @@ func newNamespacedStore(oc *clientv3.Client, conf *Config) (*GDStore, error) {
 
 	return &GDStore{*conf, kv, lease, watcher, session, oc, nil, namespaceKey}, nil
 }
+
+//Get wrapper function to call store Get operation with timeout if context is context.TODO()
```
Comment (prashanthpai) [inline glusterd2/store/store.go]: The comment for these can be better:

`Get is a wrapper function that calls clientv3.KV.Get with a default timeout if an empty context is passed`

You can do the same for `Put` and `Delete`

Comment (Madhu-1) [reply glusterd2/store/store.go]: done

File: glusterd2/store/store.go (hunk 2 above)
Comment (Madhu-1) [reply glusterd2/store/store.go]: done

File: glusterd2/peer/store-utils.go (hunk 1 above)
Comment (Madhu-1) [reply glusterd2/peer/store-utils.go]: done

File: glusterd2/store/store.go [hunk 4]
```diff
@@ -18,7 +19,10 @@ import (
 )
 
 const (
-	sessionTTL = 30 // used for etcd mutexes and liveness key
+	sessionTTL    = 30 // used for etcd mutexes and liveness key
+	getTimeout    = 5 * time.Second
```
Comment (prashanthpai) [inline glusterd2/store/store.go]: I'm sorry, I should've noticed this before. Define this as constant and move the `time.Second` to place where it's used.

```go
context.WithTimeout(context.Background(), getTimeout * time.Second)
```

**Comment by prashanthpai** [comment]
retest this please

**Pull request merged**